### PR TITLE
Add new chains Canto, Canto Testnet, Meld, Meld Testnet, Metal L2

### LINKED ISSUE #320: Add new chains for `CSUR-2024-05-28`
- [x] metal-l2 [bridge](https://bridge.metall2.com) - [whitepaper](https://cdn.metallicus.com/Metal_L2_Proposal.pdf)
- [x] canto [bridge](https://canto.io/bridge) - [docs](https://docs.canto.io/user-guides/connecting-to-canto) - EIP-155
- [x] canto-testnet - [faucet](https://www.cantofaucet.com/) - EIP-155
- [x] meld - [docs](https://dev.meld.com/develop/basics) - [bridge](https://app.meld.com/wallet)
- [x] meld testnet - [faucet](https://faucet.meld.com/)

### PR DESCRIPTION
**Pull request by hiletmis**
Closes #320 

**canto - [bridge](https://canto.io/bridge) - [docs](https://docs.canto.io/user-guides/connecting-to-canto)**
---
This chain does not allow legacy transaction, thus deterministic deployment is not available

[OwnableCallForwarder](https://www.oklink.com/canto/address/0xd5CaB12D597670e098322eD27D32659F59592349/contract)
[Api3ServerV1](https://www.oklink.com/canto/address/0x8FD1EFcB673D1438CB153783AcE6b65b7eBa1A60/contract)
[DataFeedProxy](https://www.oklink.com/canto/address/0x169101eF0fC57725cbe64d68e6454b4dFb1AC84a/contract)
[DapiProxy](https://www.oklink.com/canto/address/0xe10EEF809D831718d42432453d13A82c48291374/contract)
[DataFeedProxyWithOev](https://www.oklink.com/canto/address/0xcE14D281d4127c3DddBE527d41F386F861188343/contract)
[DapiProxyWithOev](https://www.oklink.com/canto/address/0xFcc75556cB7ca2cCfa746FA41b998F2cBaecB335/contract)
[Api3Market](https://www.oklink.com/canto/address/0x1D00656EefCE0ec4c9A45e62F1a979Be17644f14/contract)

**canto-testnet - [docs](https://docs.canto.io/evm-development/testnet)**
---
This chain does not allow legacy transaction, thus deterministic deployment is not available. Contract verification API may fail on some occasions while fetching contract code.

[OwnableCallForwarder](https://testnet.tuber.build/address/0xbf660585eFb7F31F68bB3375937b7C71376de6c8/contracts#address-tabs)
[Api3ServerV1](https://testnet.tuber.build/address/0x1B127098D19A3D6A0417560fD7df2b927FafD933/contracts#address-tabs)
[DataFeedProxy](https://testnet.tuber.build/address/0x7B4Bb6c9c13F9B70f86a187827AbC4D3186F16Fa)
[DapiProxy](https://testnet.tuber.build/address/0x6721e147874205AFCc53514a2D7233aE9caaE668)
[DataFeedProxyWithOev](https://testnet.tuber.build/address/0x4Dc2c7393Fb238F0F86403800271758fd4294a84)
[DapiProxyWithOev](https://testnet.tuber.build/address/0x47C536b505a7ab7cB4f651228B0841950Be9dE61)
[Api3Market](https://testnet.tuber.build/address/0x83B1FE7E5708d5Bd3836dE3ae83e5A2c1b92850c)

**meld - [docs](https://dev.meld.com/develop/basics) - [bridge](https://app.meld.com/wallet)**
---
This chain does not utilize a contract verification API although it is a blockscout explorer. Contracts can be verified via sourcify but that does not reflect block explorer. [Verification](https://dev.meld.com/develop/deploy-a-smart-contract/verify-your-smart-contract) in this chain has been made manually.

**meld-testnet - [faucet](https://faucet.meld.com/)**
---
This chain does not utilize a contract verification API although it is a blockscout explorer. Contracts can be verified via sourcify but that does not reflect block explorer. [Verification](https://dev.meld.com/develop/deploy-a-smart-contract/verify-your-smart-contract) in this chain has been made manually.

**metal - [bridge](https://bridge.metall2.com) - [whitepaper](https://cdn.metallicus.com/Metal_L2_Proposal.pdf)**
---
[OwnableCallForwarder](https://explorer.metall2.com/address/0xef455205966486F52f7951ba4C3517239B036835)
[Api3ServerV1](https://explorer.metall2.com/address/0x3ee097B82E270d82f829e5742cC16B7686247Ff2)
[DataFeedProxy](https://explorer.metall2.com/address/0x9Cf90E5B9292a82B214925f0b707779E0aFa0F74)
[DapiProxy](https://explorer.metall2.com/address/0x73Eb8A060e52efC2513861282Ff62B9182daC5c6)
[DataFeedProxyWithOev](https://explorer.metall2.com/address/0x18d6F2b16729F2B06eBa584B3Cd9B30771Ceb5c7)
[DapiProxyWithOev](https://explorer.metall2.com/address/0x62AbEfcFD9451364E4Eb8bC7aEF9540378e3Eb3c)
[Api3Market](https://explorer.metall2.com/address/0xE20544835208AA8aDa06BCEb666a5630dE4dAF7C)

### CODE REVIEW

File: chains/metal.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  "alias": "metal",
+  "decimals": 18,
+  "explorer": {
+    "api": {
+      "key": {
+        "required": false
+      },
+      "url": "https://explorer.metall2.com/api"
+    },
+    "browserUrl": "https://explorer.metall2.com/"
```
Comment (bdrhn9) [inline chains/metal.json:11]: Note to myself: It's op-stack, that's why [GasPriceOracle](https://explorer.metall2.com/address/0x420000000000000000000000000000000000000F) exists. However it's better to check whether this contract is being actively utilized.

File: .changeset/curvy-laws-serve.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+"@api3/chains": minor
+---
+
+Adds following chains;
```
Comment (bdrhn9) [inline .changeset/curvy-laws-serve.md:5]: Using colon `:` more appropriate to introduce list of contents.

File: chains/canto.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      "url": "https://www.oklink.com/api/explorer/v1/contract/verify/async/api/canto"
+    },
+    "browserUrl": "https://www.oklink.com/canto/"
+  },
+  "id": "7700",
+  "name": "Canto",
+  "providers": [
+    {
+      "alias": "default",
+      "rpcUrl": "https://canto.slingshot.finance/"
```
[the remark this answers is not in the document]
Comment (hiletmis) [reply chains/canto.json:18]: As paid or direct RPC url?

[the remark this answers is not in the document]
Comment (bdrhn9) [reply chains/canto.json:18]: These are direct URLs, I obtained them from [here](https://docs.canto.io/user-guides/connecting-to-canto#alternative-rpc-urls).

File: chains/canto.json [hunk 4]
```diff
@@ -16,6 +16,14 @@
     {
       "alias": "default",
       "rpcUrl": "https://canto.slingshot.finance/"
+    },
+    {
+      "alias": "public",
```
Comment (bdrhn9) [inline chains/canto.json:21]: I'd use `ansybl` as alias.

File: chains/canto.json [hunk 5]
```diff
@@ -16,6 +16,14 @@
...
+      "rpcUrl": "https://canto-rpc.ansybl.io/"
+    },
+    {
+      "alias": "alternative",
```
Comment (bdrhn9) [inline chains/canto.json:25]: `gravitychain` would be more proper.

**Review by bdrhn9 — APPROVED** [review]

**Pull request merged**